site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md

```diff
@@ -0,0 +1,469 @@
+
+Now let's break the extension to learn where to locate errors. One way to do this is to delete the ".js" from the `'./sw-omnibox.js'` import in the `service-worker.js` file. Chrome will be unable to register the service worker.
+
+Go back to chrome://extensions and refresh the extension. The following error will appear:
+
+{% Video src="video/BhuKGJaIeLNPW9ehns59NfwqKxF2/AbMNDSbURLKjH1Jm1C9Q.mp4", width="400", height="477", autoplay="true", muted="true", loop="true" %}
+
+See [Debugging extensions](tbd) for more ways debug the extension service worker.
+
+{% Aside 'caution' %}
+Don't forget to fix the file name before moving on!
+{% endAside %}
+
+### Step 4: Initialize the state {: #step-4 }
+
+Extensions can save initial values to storage on installation. To use the [`chrome.storage`][api-storage] API, we need to request permission in the manifest:
+
+{% Label %}manifest.json:{% endLabel %}
+
+```json
+{
+  ...
+  "permissions": ["storage"],
+}
+```
+
+The following code save the default suggestions to storage when the extension is first installed:
+
+{% Label %}sw-omnibox.js:{% endLabel %}
+
+```js
+...
+// Save default API suggestions
+chrome.runtime.onInstalled.addListener(({ reason }) => {
+  if (reason === 'install') {
+    chrome.storage.local.set({
+      apiSuggestions: ['tabs', 'storage', 'scripting']
+    });
+  }
+});
+```
+
+Service workers do not have direct access to the [window object][mdn-window], therefore cannot use
+[window.localStorage()][mdn-local-storage] to store values. Also, service workers are short-lived execution environments;
+they get terminated repeatedly throughout a user's browser session, which makes it incompatible with
+global variables.
+
+See [Saving state](TBD) to learn about storage options for extension service workers.
+
+### Step 5: Register your events {: #step-5 }
+
+All event listeners need to be registered in the global scope of the service worker. In other words, event listeners should not be nested in functions. This way Chrome can immediately invoke all event handlers, even if the extension's async startup logic hasn't finished. 
+
+To use the [`chrome.omnibox`][api-omnibox] API first add the omnibox keyword to the manifest:
+
+{% Label %}manifest.json:{% endLabel %}
+
+```json
+{
+  ...
+  "minimum_chrome_version": "102",
+  "omnibox": {
+    "keyword": "api"
+  },
+}
+```
+
+{% Aside 'important' %}
+The [`"minimum_chrome_version"`][manifest-min-version] explains how this key behaves when a user tries to install your extension but isn't using a compatible version of Chrome.
+
+{% endAside %}
+
+The following code registers the omnibox event listeners at the top level of the script and updates [chrome.storage][api-storage] with the most recent api search.
```

Comment:
The code also creates suggestions. Maybe add a bit more explanation and maybe even split up the code.

Reply:
Good idea. On it! 🏃🏻‍♀️💨